rotkehlchen/data/binance_import_utils.py

```diff
@@ -0,0 +1,366 @@
+                    return []
+                row['usd_value'] = row['Change'] * price
+
+        # Group rows depending on whether they are fee or not and then sort them by amount
+        rows_grouped_by_fee: Dict[bool, List[BinanceCsvRow]] = defaultdict(list)
+        for row in data:
+            is_fee = row['Operation'] == 'Fee'
+            rows_grouped_by_fee[is_fee].append(row)
+
+        for rows_group in rows_grouped_by_fee.values():
+            rows_group.sort(key=lambda x: x['Change'] if same_assets else x['usd_value'], reverse=True)  # noqa: E501
+
+        # Grouping by combining the highest sold with the highest bought and the highest fee
+        # Using fee only we were provided with fee (checking by "True in rows_by_operation")
+        grouped_trade_rows = []
+        while len(rows_grouped_by_fee[False]) > 0:
+            cur_batch = [rows_grouped_by_fee[False].pop(), rows_grouped_by_fee[False].pop(0)]
+            if True in rows_grouped_by_fee:
+                cur_batch.append(rows_grouped_by_fee[True].pop())
+            grouped_trade_rows.append(cur_batch)
+
+        # Creating trades structures based on grouped rows data
+        raw_trades: List[Trade] = []
+        for trade_rows in grouped_trade_rows:
+            to_asset: Optional[Asset] = None
+            to_amount: Optional[AssetAmount] = None
+            from_asset: Optional[Asset] = None
+            from_amount: Optional[AssetAmount] = None
+            fee_asset: Optional[Asset] = None
+            fee_amount: Optional[Fee] = None
+            trade_type: Optional[TradeType] = None
+
+            for row in trade_rows:
+                cur_asset = row['Coin']
+                amount = row['Change']
+                if row['Operation'] == 'Fee':
+                    fee_asset = cur_asset
+                    fee_amount = Fee(amount)
+                else:
+                    trade_type = TradeType.SELL if row['Operation'] == 'Sell' else TradeType.BUY  # noqa:  E501
+                    if amount < 0:
+                        from_asset = cur_asset
+                        from_amount = AssetAmount(-amount)
+                    else:
+                        to_asset = cur_asset
+                        to_amount = amount
+
+            # Validate that we have received proper assets and amounts.
+            # There can be no fee, so we don't validate it
+            if (
+                to_asset is None or from_asset is None or trade_type is None or
+                to_amount is None or to_amount == ZERO or
+                from_amount is None or from_amount == ZERO
+            ):
+                log.warning(
+                    f'Skipped binance rows {data} because '
+                    f'it didn\'t have enough data',
+                )
+                db.msg_aggregator.add_warning('Skipped some rows because couldn\'t find amounts or it was zero')  # noqa: E501
+                continue
+
+            rate = FVal(to_amount) / FVal(from_amount)
```

Comment:
Why are you invoking the FVal constructor on two things that are already FVals here?

Reply:
No reason to do it here indeed. Perhaps it was used in the past when `to_amount` and `from_amount` were not `FVal`s but dict keys.